src/NSwag.SwaggerGeneration.AspNetCore/OperationParameterProcessor.cs

```diff
@@ -0,0 +1,334 @@
+                    operationParameter.IsRequired = true; // Path is always required => property not needed
+
+                    if (_settings.SchemaType == SchemaType.Swagger2)
+                        operationParameter.IsNullableRaw = false;
+
+                    context.OperationDescription.Operation.Parameters.Add(operationParameter);
+                }
+                else
+                {
+                    var parameterInfo = _settings.ReflectionService.GetDescription(apiParameter.Type, parameter.GetCustomAttributes(), _settings);
+                    if (await TryAddFileParameterAsync(context, parameterInfo, parameter).ConfigureAwait(false) == false)
+                    {
+                        if (apiParameter.Source == BindingSource.Header)
+                        {
+                            var operationParameter = await context.SwaggerGenerator.CreatePrimitiveParameterAsync(parameterName, parameter).ConfigureAwait(false);
+                            operationParameter.Kind = SwaggerParameterKind.Header;
+
+                            context.OperationDescription.Operation.Parameters.Add(operationParameter);
+                        }
+                        else if (apiParameter.Source == BindingSource.Query)
+                        {
+                            var operationParameter = await context.SwaggerGenerator.CreatePrimitiveParameterAsync(parameterName, parameter).ConfigureAwait(false);
+                            operationParameter.Kind = SwaggerParameterKind.Query;
+
+                            context.OperationDescription.Operation.Parameters.Add(operationParameter);
+                        }
+                        else
+                        {
+                            if (apiParameter.Source == BindingSource.Body)
+                                await AddBodyParameterAsync(context, parameterName, parameter).ConfigureAwait(false);
+
+                            else if (parameterInfo.IsComplexType)
+                            {
+                                // Try to find a [WillReadBody] attribute on either the action parameter or the bindingAttribute's class
+                                var willReadBodyAttribute = attributes.TryGetIfAssignableTo("WillReadBodyAttribute", TypeNameStyle.Name);
+
+                                if (willReadBodyAttribute == null)
+                                    await AddBodyParameterAsync(context, parameterName, parameter).ConfigureAwait(false);
+                                else
+                                {
+                                    // Try to get a boolean property value from the attribute which explicity tells us whether to read from the body
+                                    // If no such property exists, then default to false since WebAPI's HttpParameterBinding.WillReadBody defaults to false
+                                    var willReadBody = willReadBodyAttribute.TryGetPropertyValue("WillReadBody", true);
```

Comment:
This all looks like Legacy

Reply:
All of the `WillReadBody`

Reply:
Oh nvm this is part of NSwag. Ignore me

Reply:
I think this is legacy: https://github.com/RSuter/NSwag/issues/634

Reply:
Ok, I'll remove looking up this attribute.